### PR TITLE
Feature/xcui lane

### PR DESCRIPTION
**Pull request by zacoid55**

### CODE REVIEW

File: Fastfile [hunk 1]
```diff
@@ -14,6 +14,15 @@ default_platform :ios
     scan(skip_slack: skip_slack)
   end
 
+  lane :ui_test do
+    _setup()
+    skip_slack = ENV['SCAN_SLACK_CHANNEL'].to_s.strip.empty?
+    scan(skip_slack: skip_slack,
+         devices: ["iPhone 8"],
```
Comment (Neil3079) [inline Fastfile]: We should make the devices this can run on configurable

Comment (KITSTABLucianoMarisi) [inline Fastfile]: is there a reason why this is not `SCAN_DEVICE`?

File: Fastfile [hunk 2]
```diff
@@ -14,6 +14,15 @@ default_platform :ios
...
+         output_types: "html",
```
Comment (Neil3079) [inline Fastfile]: Again worth the output type being configurable, its fine to have a default but people should be able to choose

File: Fastfile [hunk 3]
```diff
@@ -14,6 +14,15 @@ default_platform :ios
...
+         scheme: ENV['UI_TEST_SCHEME'])
```
Comment (Neil3079) [inline Fastfile]: following naming convention for env variables we have defined our selves can we update this to `UI_TEST_SCHEME `

File: Fastfile (hunk 1 above)
Comment (lucianomarisi) [inline Fastfile]: is there a reason why this is not `SCAN_DEVICE`?

**Review by Neil3079 — COMMENTED** [review]
Could we add to the changelog as this will have to be a new version tag

File: Fastfile (hunk 3 above)
Comment (lucianomarisi) [reply Fastfile]: did you mean `TAB_UI_TEST_SCHEME`?

File: Fastfile [hunk 4]
```diff
@@ -14,6 +14,15 @@ default_platform :ios
     scan(skip_slack: skip_slack)
   end
 
+  lane :ui_test do
+    _setup()
+    skip_slack = ENV['SCAN_SLACK_CHANNEL'].to_s.strip.empty?
+    scan(skip_slack: skip_slack,
+         devices: ['TAB_UI_TEST_DEVICES'],
+         output_types: ['TAB_OUTPUT_TYPES'],
```
Comment (Neil3079) [inline Fastfile]: Could this be a more descriptive name, Output Types could refer to a lot of different things

File: Fastfile [hunk 5]
```diff
@@ -117,6 +126,12 @@ default_platform :ios
     if is_ci && ENV['TAB_XCODE_PATH'] != nil
       xcode_select(ENV['TAB_XCODE_PATH'])
     end
+    if ENV['TAB_UI_TEST_SCHEME'] != nil && ENV['TAB_OUTPUT_TYPES'] == nil
```
Comment (Neil3079) [inline Fastfile:129]: This feels like it belongs in setup not deploy to hockey

Comment (zacoid55) [reply Fastfile:129]: @Neil3079 it is in setup isn't it?

Comment (Neil3079) [reply Fastfile:129]: It is my bad!!!

**Review by oliver-dew — APPROVED** [review]

**Review by KaneCheshire — APPROVED** [review]

**Pull request merged**